From 4ea4864ec748ad77767bd02b26c76cb9d61604fe Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Thu, 17 Mar 2022 14:11:48 +0200
Subject: [PATCH 1/3] plat: nxp/s32: Force boot_info.c regeneration

This aims to prevent the reusage of the generated boot_info.c
when switching between branches.

Issue: ALB-8686
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 plat/nxp/s32/s32_common.mk | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/plat/nxp/s32/s32_common.mk b/plat/nxp/s32/s32_common.mk
index 7f76d5ed2..c6cf0c2ee 100644
--- a/plat/nxp/s32/s32_common.mk
+++ b/plat/nxp/s32/s32_common.mk
@@ -255,7 +255,7 @@ ${DUMMY_STAGE}: | ${BUILD_PLAT}
 	${Q}${ECHO} "dummy_stage" > $@
 
 # Replace all fiptool args with a dummy state except '--align' parameter
-${DUMMY_FIP}: fiptool ${DUMMY_STAGE} | ${BUILD_PLAT}
+${DUMMY_FIP}: fiptool ${DUMMY_STAGE} FORCE | ${BUILD_PLAT}
 	${Q}${ECHO} "  FIP     $@"
 	${Q}ARGS=$$(echo "${FIP_ARGS}" | sed "s#\(--[^ a]\+\)\s\+\([^ ]\+\)#\1 ${DUMMY_STAGE}#g"); \
 		${FIPTOOL} create $${ARGS} "$@_temp"
@@ -361,7 +361,7 @@ ${BL2_W_DTB}: bl2 dtbs ${DTB_SIZE_FILE}
 	@cp ${BUILD_PLAT}/fdts/${DTB_FILE_NAME} $@
 	@dd if=${BUILD_PLAT}/bl2.bin of=$@ seek=$$(printf "%d" ${DTB_SIZE}) status=none oflag=seek_bytes
 
-${BOOT_INFO_SRC}: ${FIP_SD_OFFSET_FILE} ${FIP_EMMC_OFFSET_FILE} ${FIP_QSPI_OFFSET_FILE} ${FIP_MEMORY_OFFSET_FILE} ${FIP_HDR_SIZE_FILE} ${DTB_SIZE_FILE}
+${BOOT_INFO_SRC}: ${FIP_SD_OFFSET_FILE} ${FIP_EMMC_OFFSET_FILE} ${FIP_QSPI_OFFSET_FILE} ${FIP_MEMORY_OFFSET_FILE} ${FIP_HDR_SIZE_FILE} ${DTB_SIZE_FILE} FORCE
 	${Q}${ECHO} "  CREATE  $@"
 	${Q}${ECHO} "const unsigned long fip_sd_offset = $$(cat ${FIP_SD_OFFSET_FILE});" > ${BOOT_INFO_SRC}
 	${Q}${ECHO} "const unsigned long fip_emmc_offset = $$(cat ${FIP_EMMC_OFFSET_FILE});" >> ${BOOT_INFO_SRC}
-- 
2.17.1

